### PR TITLE
Update to ClimaAtmos v0.11.0 and ClimaLSM v0.2.3

### LINKED ISSUE #292: Update `ClimaAtmos` to v0.11.0 in `experiments/AMIP/modular/Project.toml`
Tracked as part of SDI #281

### PR DESCRIPTION
**Pull request by valeriabarra**
## Purpose 
The Purpose of this PR is to update the `experiments/AMIP/modular` environment to use the latest ClimaAtmos release, `v0.10.0`.

In order for CI integration tests to work, I also had to update the `experiments/AMIP/moist_mpi_earth` environment.

Closes #292 , #297 and #298 

## Content
- Modified `compat` entries in the top-level env, the `experiments/AMIP/modular` env, and the `experiments/AMIP/moist_mpi_earth` environments  to be able to update using the Package manager
- Updated to the latest `ClimaLSM` v0.2.3 release 
- Updated to the latest `ClimaAtmos` v0.11.0 release 
- Updated the `ClimaAtmos`-related CLI options
- Updated to the most recent `ClimaCore`, `ClimaLSM`, and `ClimaAtmos` API

Review checklist

I have:
- followed the codebase contribution guide: https://clima.github.io/ClimateMachine.jl/latest/Contributing/
- followed the style guide: https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/
- followed the documentation policy: https://github.com/CliMA/policies/wiki/Documentation-Policy
- checked that this PR does not duplicate an open PR.

In the Content, I have included 
- relevant unit tests, and integration tests, 
- appropriate docstrings on all functions, structs, and modules, and included relevant documentation.

- [x] I have read and checked the items on the review checklist.

### CODE REVIEW

**Review by charleskawczynski — COMMENTED** [review]
We'll need to add something like

```
      - echo "--- Instantiate amip modular env"
      - "julia --project=experiments/AMIP/modular/ -e 'using Pkg; Pkg.instantiate(;verbose=true)'"
      - "julia --project=experiments/AMIP/modular/ -e 'using Pkg; Pkg.precompile()'"
      - "julia --project=experiments/AMIP/modular/ -e 'using Pkg; Pkg.status()'"
```
to the buildkite yaml for CI to work. I'm actually not sure how that environment is activated at the moment

File: experiments/AMIP/modular/Project.toml [hunk 1]
```diff
@@ -52,16 +52,15 @@ ArgParse = "1.1"
...
 CLIMAParameters = "0.4, 0.5, 0.6, 0.7"
-ClimaAtmos = "0.9"
+ClimaAtmos = "0.10"
 ClimaComms = "0.3"
 ClimaCommsMPI = "0.3"
 ClimaCore = "0.10"
 ClimaCorePlots = "0.2"
 ClimaCoreTempestRemap = "0.3"
 ClimaCoupler = "0.1"
-ClimaLSM = "0.2"
```
Comment (juliasloan25) [inline experiments/AMIP/modular/Project.toml:62]: Why did you remove the ClimaLSM compat?

**Review by juliasloan25 — COMMENTED** [review]
Looks good overall. I have one concern though - are you planning to merge this into main as is? It might be better to resolve the failing tests/CI before merging.

**Comment by valeriabarra** [comment]
> Looks good overall. I have one concern though - are you planning to merge this into main as is? It might be better to resolve the failing tests/CI before merging.

Hi Julia, thank you. No, absolutely. The idea is not to merge this as is. Of course I want/need to fix CI. I am just also working on other things so I haven't back to this yet. Hopefully this afternoon.

**Comment by valeriabarra** [comment]
> We'll need to add something like
> 
> ```
>       - echo "--- Instantiate amip modular env"
>       - "julia --project=experiments/AMIP/modular/ -e 'using Pkg; Pkg.instantiate(;verbose=true)'"
>       - "julia --project=experiments/AMIP/modular/ -e 'using Pkg; Pkg.precompile()'"
>       - "julia --project=experiments/AMIP/modular/ -e 'using Pkg; Pkg.status()'"
> ```
> 
> to the buildkite yaml for CI to work. I'm actually not sure how that environment is activated at the moment

It looks like in [CI](https://github.com/CliMA/ClimaCoupler.jl/blob/main/.buildkite/pipeline.yml#L41-L44) we're instantiating the other environment `experiments/AMIP/moist_mpi_earth`. Should I try to use this updated `experiments/AMIP/modular` in there instead?

**Comment by valeriabarra** [comment]
@charleskawczynski do you know what is going on with the [Ubuntu-latest](https://github.com/CliMA/ClimaCoupler.jl/actions/runs/4833874439/jobs/8614420122?pr=293) action?

**Comment by charleskawczynski** [comment]
> @charleskawczynski do you know what is going on with the [Ubuntu-latest](https://github.com/CliMA/ClimaCoupler.jl/actions/runs/4833874439/jobs/8614420122?pr=293) action?

It looks like a ClimaLSM error:

```julia
home/runner/work/ClimaCoupler.jl/ClimaCoupler.jl/test/conservation_checker_tests.jl:120
  Got exception outside of a @test
  MethodError: no method matching (ClimaLSM.Parameters.LSMParameters{Float64, Float64, Float64})(::Float64, ::Float64, ::Float64, ::Float64, ::Float64, ::Float64, ::Float64, ::Float64, ::Float64, ::Float64, ::Float64, ::Float64, ::Float64, ::Float64, ::Float64)
  Stacktrace:
    [1] macro expansion
      @ ~/work/ClimaCoupler.jl/ClimaCoupler.jl/test/conservation_checker_tests.jl:130 [inlined]
...
```

**Comment by charleskawczynski** [comment]
Probably the simplest thing to do, in this particular situation, would be to use their utility:
 - At the top of this test directory, `include(joinpath(pkgdir(ClimaLSM), "parameters", "create_parameters.jl"))`
 - Then use `earth_param_set = create_lsm_parameters(FT)` (this function is [here](https://github.com/CliMA/ClimaLSM.jl/blob/5c42544f032e04af80b5a29f25e23011bcdf6769/parameters/create_parameters.jl#L13)) where it's breaking. There's probably been breaking changes to their internal parameter set.

**Comment by valeriabarra** [comment]
> Probably the simplest thing to do, in this particular situation, would be to use their utility:
> 
> * At the top of this test directory, `include(joinpath(pkgdir(ClimaLSM), "parameters", "create_parameters.jl"))`
> * Then use `earth_param_set = create_lsm_parameters(FT)` (this function is [here](https://github.com/CliMA/ClimaLSM.jl/blob/5c42544f032e04af80b5a29f25e23011bcdf6769/parameters/create_parameters.jl#L13)) where it's breaking. There's probably been breaking changes to their internal parameter set.

I fixed it. It was a matter of updating the `LSMParameters` call, because its API had changed.

File: experiments/AMIP/modular/Project.toml (hunk 1 above)
Comment (valeriabarra) [reply experiments/AMIP/modular/Project.toml:62]: It slipped. Thanks for the catch. I put it back

**Comment by valeriabarra** [comment]
This PR is ready for review! 
Tests seem to be passing now. 
Once approved, I'll reduce commits. Thanks all

**Comment by LenkaNovak** [comment]
Hi Valeria, thank you for this! To help with the crashing Buildkites, we’ll need to update the `cli_options` file by copying the Atmos CLI options [here](https://github.com/CliMA/ClimaAtmos.jl/blob/main/examples/hybrid/cli_options.jl#L7-L348) to under the [ClimaAtmos flags](https://github.com/CliMA/ClimaCoupler.jl/blob/valeria/atmos-v0.10/experiments/AMIP/modular/cli_options.jl#L34)

**Comment by valeriabarra** [comment]
> Hi Valeria, thank you for this! To help with the crashing Buildkites, we’ll need to update the `cli_options` file by copying the Atmos CLI options [here](https://github.com/CliMA/ClimaAtmos.jl/blob/main/examples/hybrid/cli_options.jl#L7-L348) to under the [ClimaAtmos flags](https://github.com/CliMA/ClimaCoupler.jl/blob/valeria/atmos-v0.10/experiments/AMIP/modular/cli_options.jl#L34)

Yes, thank you. I realized that after I wrote my comment. I'll get to it on Monday. I think we're close!

**Comment by valeriabarra** [comment]
The issue was identified to be in the logic ClimaAtmos tries to prioritize parsed args and populate the toml file (see this [line ](https://github.com/CliMA/ClimaAtmos.jl/blob/270a96f38ceb54d73f01c363cbaff297278eb782/examples/hybrid/parameter_set.jl#L56)). Once this is fixed in ClimaAtmos, I'll propagate the change here in the Coupler.

File: experiments/AMIP/modular/components/flux_calculator.jl [hunk 2]
```diff
@@ -12,7 +12,7 @@ of the lowest level.
 function set_ρ_sfc!(ρ_sfc, T_S, integrator)
     ts = integrator.p.ᶜts
     thermo_params = CAP.thermodynamics_params(integrator.p.params)
-    ts_int = Spaces.level(ts, 1)
+    ts_int = ClimaCore.Spaces.level(ts, 1)
```
Comment (charleskawczynski) [inline experiments/AMIP/modular/components/flux_calculator.jl:15]: Being that there are several key modules in ClimaCore, it may be easier (result in a smaller diff, too) to add `import ClimaCore.Spaces as Spaces` to the top of wherever this is included.

**Comment by valeriabarra** [comment]
YAY! CI finally [passed](https://buildkite.com/clima/climacoupler-ci/builds/1190#_). 

@juliasloan25 , @charleskawczynski if you would like to give a final review, then if approved, I'll go ahead with squashing/reducing commits. Thanks everybody!

**Review by juliasloan25 — COMMENTED** [review]
looks good! Just a few small comments

File: experiments/AMIP/modular/Project.toml [hunk 3]
```diff
@@ -46,22 +46,22 @@ TerminalLoggers = "5d786b92-1e48-4d6f-9151-6b4477ca9bed"
 Test = "8dfed614-e22c-5e08-85e1-65c5234f0b40"
 Thermodynamics = "b60c26fb-14c3-4610-9d3e-2d17fe7ff00c"
 UnPack = "3a884ed6-31ef-47d7-9d2a-63182c4928ed"
+YAML = "ddb6d928-2868-570f-bddf-ab3f9cf99eb6"
```
[the remark this answers is not in the document]
Comment (valeriabarra) [reply experiments/AMIP/modular/Project.toml:49]: It was needed at some point while I was going through the update. Otherwise it would throw an error. Let me try to remove it now and test it again.

File: experiments/AMIP/moist_mpi_earth/coupler_utils/flux_calculator.jl [hunk 4]
```diff
@@ -48,18 +48,16 @@ function calculate_surface_fluxes_atmos_grid!(integrator, info_sfc)
...
     Fields.bycolumn(axes(Y.c.uₕ)) do colidx
         get_surface_fluxes!(Y, p, colidx)
         # corrections (accounting for inhomogeneous surfaces)
-        @. p.dif_flux_energy_bc[colidx] = # todo: get rid - shouldn't make any difference anyway
-            Geometry.WVector(correct_e_over_ice(p.surface_conditions[colidx], ice_mask[colidx]))
-        @. p.dif_flux_ρq_tot_bc[colidx] =
-            Geometry.WVector(correct_q_over_ice(p.surface_conditions[colidx], ice_mask[colidx]))
-
+        @. p.ρ_dif_flux_h_tot[colidx] = # checking right quantity in ClimaAtmos v0.11.0
+            -Geometry.WVector(correct_e_over_ice(p.surface_conditions[colidx], ice_mask[colidx]))
```
[the remark this answers is not in the document]
Comment (valeriabarra) [reply experiments/AMIP/moist_mpi_earth/coupler_utils/flux_calculator.jl:56]: But the experiments in the `moist_mpi_earth` environment seem to still use `ice_mask` (see this [line](https://github.com/CliMA/ClimaCoupler.jl/blob/e62b5738322a0831af6e9c8e2213ed2ae532be93/experiments/AMIP/moist_mpi_earth/coupler_utils/flux_calculator.jl#LL57C17-L57C17) on `main`). I think we should keep it then.

**Comment by valeriabarra** [comment]
Bors r+
